### ISSUE TITLE
Disabling Script Editor also disable the ability to create "New Scripts" from inspector

### ISSUE THREAD

**Issue by AndreSacilotto** [godotengine/godot#73702]
### Godot version

3.5.1 and 4.0.RC2

### System information

Windows

### Issue description

When disabling the script editor you also lose the ability to create new scripts from nodes.

Why I disabled it?
I use c# on external editor, and some times I missclick and open on the script tab, and its saving behavior is weird and it sometimes (rarely) override my changes. So I decided to disable it, but for my surprise it disable the ability to create new scripts.

Why it's a bug?
> Script Editor:
> Allows to edit scripts using the integrated script editor.

The disabling the script feature dont say that "you will be disabled of creating new scripts from inspector". 
I stayed 1 week thinking it has just a bug with the new godot version, but nope it also exist on 3.5.

### Steps to reproduce

1. Editor -> Manage Editor Features
2. Disable Script Editor
3. Select any node  right click on script -> New Script
4. Nothing will happen

### Minimal reproduction project

Editor bug

**Comment by AndreSacilotto** [comment]
I forgot to mention that it also disables the ability to open the script on the external editor by double-clicking the file in the filesystem.

**Comment by nongvantinh** [comment]
The reason is when someone wants to **_disable_** the _scripting future_, it also disallows you from interacting with it.

Disabling some future means you will not use them anymore, and because that future causes distraction that would lead to less productivity.

For instance, an artist might want to work with the scene only, a teacher thinks that hide some future can help students fucus more on the topic, ...

**Comment by nongvantinh** [comment]
If you really want to disable Scripting Editor's future but still want to use some functionalities of it. 

Seeing you currently using a RC version, and I have made a PR for this, you could pull [that branch](https://github.com/nongvantinh/godot/tree/fix-73702) to your PC and compile it yourself.

And from now on, whenever you need some future that is specific to your need. You can modify the source code and implement it yourself. That's how companies do.

**Comment by AndreSacilotto** [comment]
I'm just documenting a probably unintended behavior, in other words reporting a bug. 

You don't need to replicate what [akien-mga](https://github.com/akien-mga) said in your PR. And you probably miss understood what he said there, he was saying that the alterations you have done breaks the actual intended behavior.

So to solve it you would actually need to preserve the old behavior while adding a new one.

A idea would be split script editor into to other options "Script Text Editor" and "Script Editing", so preserving the old behavior while adding a new one.
Idk if it bothers me enough to try to implement it or not, I will see.

**Comment by nongvantinh** [comment]
You're taking it personally.

> Disabling some future means you will not use them anymore

If the design was to _**disallow the usage of that future**_, then why would we add something that will break the design and the abundant code added only benefit a few people?

Because you were reported a bug, and I feel I can **fix** it so I create a PR for this issue without knowing the original design was to disable that future completely.

I consider your case, you  

> some times I missclick

and cause the editor to switch to the scripting tab was not a big deal. 

And the modification only benefit for you, so I explained more about the design for you and I also guide you to use my PR for your needs and if you need more, you can modify it yourself.

**Comment by AndreSacilotto** [comment]
@akien-mga 
Could you point out where that use case is explicitly said in the docs I can't find it.

What you mean by future are you misspelling feature? Or is some kind of term?

A issue not always need to resolved with a PR to the engine some times is just missing documentation.

But of course is personal, every issue is, you want to solve it, because it's a problem for you. And maybe more people tried the same thing and simply notice that don't work as they expected and have move on.

> and cause the editor to switch to the scripting tab was not a big deal.

Shame on me, I have written a unrelated issue as exemple my bad, I will fix that.

**Comment by nongvantinh** [comment]
It's 1 AM on my side, and I've been working all day long to care more about grammar mistake.
![image](https://user-images.githubusercontent.com/53887662/220718297-f433944d-e4cd-4507-a631-667c0b6a4d6b.png)

It was my fault for typing "future". when it means feature. Just as _i_ means _j_

We don't discuss grammar mistake here as we are not native speakers and I also notice yours. Let's get back to the base case:

1. You report a bug, it labeled **bug** and it means _**describes something that is not working properly**_.
2. A bug needs to be fixed, and I help the developer fix it by creating a PR of my own free will so they can spend their time on more important task.
3. Until akien-mga point out this is intended behaviour, so I guide you to use it for your needs.

**Comment by nongvantinh** [comment]
It's 1 AM on my side, and I've been working all day long to care more about grammar mistake.
![image](https://user-images.githubusercontent.com/53887662/220722904-81436846-bc46-40dc-880d-07ab240031eb.png)

It was my fault for typing "future". when it means feature. Just as _i_ means _j_

We don't discuss grammar mistake here as we are not native speakers and I also notice yours. Let's get back to the base case:

1. You report a bug, it labeled **bug** and it means _**describes something that is not working properly**_.
2. A bug needs to be fixed, and I help the developer fix it by creating a PR of my own free will so they can spend their time on more important task.
3. Until akien-mga point out this is intended behaviour, so I guide you to use it for your needs.

**Comment by akien-mga** [comment]
Please keep the discussion focused on the issue. There's a bad vibe in the last interactions which seems completely unnecessary.

Now, this is no way a priority and we're not going to merge any PR for this issue during release candidate stage for 4.0, so there's no need to rush things.

I pointed out a potential difference in expectations for what this should do. It can be discussed peacefully, and there's no rush.

**Comment by Zireael07** [comment]
@akien-mga I think I saw a comment by you explaining two different expectations, but I don't see it here? Were there two threads or something?

Personally, if the design *is* to disable creating scripts entirely, the option should clearly say so both in the settings and in docs. Tbh, I am thinking that the setting should be renamed to "disable scripts" and the current "disable script editor" would then work for the "I want to use external editor only" use case. (And both settings should be a thing!)

**Comment by akien-mga** [comment]
Yes, I commented on the PR: https://github.com/godotengine/godot/pull/73760#issuecomment-1440057564

I'll repeat it here:

The main question is what is the use case for disabling the script editor. There seems to be conflicted use cases:

- Disabling all scripting functionality (e.g. for teachers who want to teach students with only nodes/inspector/animations/premade custom nodes but no scripting initially). That's what the current implementation was intended for.
- Hiding the script editor because users want to use an external editor, and thus remove the internal editor from the interface. This is what this PR seems to allow, but breaks the above requirement.
